force-app/main/default/classes/DummyJSONCallout.cls

```diff
@@ -66,16 +70,28 @@ public with sharing class DummyJSONCallout {
 	@TestVisible // Allows test class to see this method. Since it is private, it would not be visible otherwise.
 	private static Contact createContactFromJson(String jsonResponse) {
 		// Deserialize the JSON string into map of primitive data types.
-
+		Map<String, Object> jsonMap = (Map<String, Object>) JSON.deserializeUntyped(jsonResponse);
 		// Create a new contact from the JSON response.
-
+		Contact cont = new Contact();
 		// Set the contact fields from the JSON response.
-
+		cont.FirstName = (String) jsonMap.get('firstName');
+		if(jsonMap.get('lastName') == null) {
+			cont.LastName = 'Unknown';
```

Comment:
Very nice checking this ahead of time, it will make sure your code doesn't break in the future